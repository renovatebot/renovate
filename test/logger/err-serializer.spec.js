import configSerializer from '../../lib/logger/err-serializer';

describe('logger/err-serializer', () => {
  it('expands errors', () => {
    const err = {
      a: 1,
      b: 2,
      message: 'some message',
      response: {
        body: 'some response body',
      },
      gotOptions: {
        headers: {
          authorization: 'Bearer abc',
        },
        auth: 'test:token',
      },
    };
    expect(configSerializer(err)).toMatchSnapshot();
  });
  it('handles missing fields', () => {
    const err = { a: 1, stack: 'foo', body: 'some body' };
    expect(configSerializer(err)).toMatchSnapshot();
  });
});
