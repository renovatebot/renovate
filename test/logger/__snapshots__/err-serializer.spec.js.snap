// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logger/err-serializer expands errors 1`] = `
Object {
  "a": 1,
  "b": 2,
  "body": "some response body",
  "gotOptions": Object {
    "headers": Object {
      "authorization": "** redacted **",
    },
  },
  "message": "some message",
  "response": Object {
    "body": "some response body",
  },
}
`;

exports[`logger/err-serializer handles missing fields 1`] = `
Object {
  "a": 1,
  "body": "some body",
  "stack": "foo",
}
`;
